services/src/main/java/org/keycloak/services/managers/ClientManager.java

```diff
@@ -366,4 +367,19 @@ private boolean showClientCredentialsAdapterConfig(ClientModel client) {
         return authenticator.getAdapterConfiguration(client);
     }
 
+    private boolean isInternalClient(String realmName, String clientId) {
+        if (!"master".equals(realmName)) {
+            return false;
+        }
+
+        final String internalClientSuffix = "-realm";
+
+        if (clientId.endsWith(internalClientSuffix)) {
+            return realmManager.getSession().realms()
+                    .getRealmByName(
+                            clientId.substring(0, clientId.length() - internalClientSuffix.length())) != null;
+        }
+
+        return defaultClients.contains(clientId);
+    }
```

Comment:
```suggestion
    private boolean isInternalClient(String realmName, String clientId) {
        if (defaultClients.contains(clientId) return true;
        
        if (!"master".equals(realmName)) {
            return false;
        }

        final String internalClientSuffix = "-realm";

        if (clientId.endsWith(internalClientSuffix)) {
            return realmManager.getSession().realms()
                    .getRealmByName(
                            clientId.substring(0, clientId.length() - internalClientSuffix.length())) != null;
        }

        return false;
    }
```

Reply:
Fixed. I also improved the testing, so it now tests the removal on every default client.